tests/transforms/test_defer_measurements.py

```diff
@@ -160,9 +160,41 @@ def qnode():
         assert isinstance(sec_ctrl_op.queue[0], qml.CRZ)
         assert sec_ctrl_op.data == [sec_par]
 
+    def test_correct_ops_in_tape_inversion(self):
```

Comment:
minor: I find it confusing that some tests that test the logic of `qml.cond()` are in `test_condition`, and others are in `test_defer_measurements` 🤔 

I wonder if in the future devs will get confused looking for all the tests.

But I can't think of a good solution

Reply:
Maybe here is one approach that could work: maybe there should be tests in `test_condition` that look similar to below, but do _not_ apply the `defer_measurements` transform - they simply check that `qml.cond()` is doing what it should be doing in different edge cases/for different inputs.

That is, check that it is generating the correct `Conditional` objects.

Reply:
What if a note is added to the module docstring in the test file?

> Maybe here is one approach that could work: maybe there should be tests in test_condition that look similar to below, but do not apply the defer_measurements transform - they simply check that qml.cond() is doing what it should be doing in different edge cases/for different inputs.
> That is, check that it is generating the correct Conditional objects.

Would this now be covered by the new unit tests checking that queuing works?

Reply:
https://github.com/PennyLaneAI/pennylane/pull/2275/commits/6109a354aca3ab67c7b1675df017a211921f0408